aten/src/ATen/native/cudnn/RNN.cpp

```diff
@@ -1025,6 +1025,7 @@ std::tuple<Tensor, Tensor, Tensor, std::vector<Tensor>> _cudnn_rnn_backward(
     std::array<bool, 4> output_mask
     ) {
 
+  check_device(input, hx, weight);
```

Comment:
You don't need device checks in backward. It should be correct by construction of the graph. They're only there to catch user errors.

Reply:
I see, I will remove the checks from backward. For the confusion, can I ask why I only need to check devices at cuDNN paths?